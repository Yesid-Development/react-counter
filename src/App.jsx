import React from "react";
import PropTypes from "prop-types";

const App = ({ greeting, subtitle }) => {
  return (
    <>
      <h1>{ greeting }</h1>
      <p>{ subtitle }</p>
    </>
  );
};

App.propTypes = {
	greeting: PropTypes.string.isRequired
};

App.defaultProps = {
	subtitle: 'Learning React with Fernando Herrera'
}

export default App;
