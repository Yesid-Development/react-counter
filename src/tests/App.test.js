import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import App from '../App';

describe('Pruebas del componente App', () => {
  test('Debe mostrar <App /> correctamente', () => {
    const greeting = 'Hola';

    const wrapper = shallow(<App greeting={greeting} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar el subtitulo enviado por props', () => {
    const greeting = 'Hola';
    const subtitle = 'I am a simple subtitle of test';

    const wrapper = shallow(<App greeting={greeting} subtitle={subtitle} />);

    const paragraph = wrapper.find('p').text();

    expect(paragraph).toBe(subtitle);

  });
});
