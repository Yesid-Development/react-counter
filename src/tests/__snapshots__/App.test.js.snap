// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente App Debe mostrar <App /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola
  </h1>
  <p>
    Learning React with Fernando Herrera
  </p>
</Fragment>
`;
